ndinterp/src/grid.rs

```diff
@@ -28,7 +28,16 @@ pub struct Grid<D: Dimension> {
     pub values: Array<f64, D>,
 }
 
-impl Grid<Ix1> {
+/// A grid slice is always 1-Dimensional
+#[derive(Debug)]
+pub struct GridSlice<'a> {
```

Comment:
Eventually, you started playing with lifetimes...

Reply:
I don't understand why the compiler is not able to deduce all of them though.

Reply:
There are rules to infer references lifetimes for methods, but even those are limited (to simplest but also most common cases, so they are very useful). You definitely want to infer as much as possible, but consistency is hard to impose when you have many rules (I recently read about attempts to introduce linear types, and they are a mess...).

I'm not sure, but I believe the reason you need to make them explicit should be that there is an alternative way of doing, e.g. having more lifetimes and some of them explicitly specified (like generics), or something like that. Maybe?